### PR TITLE
Add InvalidOptionError Exception

### LINKED ISSUE #2: Raise specific exceptions on 400
For example, pass `InvalidOption` into `chart` and you get a 400 with a JSON response. Raise something better than a generic `Faraday::ClientError: the server responded with status 400`.

### PR DESCRIPTION
**Pull request by rodolfobandeira**
close #2 

Raise an `InvalidOption` Exception if an invalid option is passed to `Chart.get` and IEX returns 400 `ClientError`

### CODE REVIEW

**Comment by rodolfobandeira** [comment]
@dblock I notice Rubocop is bitching about some rules that aren't actually failing on my local. What should we do? Create a custom local `rubocop.yml`?

**Comment by dblock** [comment]
For Robocop run `rubocop -a ; rubocop --auto-gen-config`.

File: lib/iex/errors/invalid_option_error.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module IEX
+  module Errors
+    class InvalidOptionError < StandardError
+      attr_reader :response
+
+      def initialize(response)
+        @response = response
+        super "Invalid option passed into your get chart hash"
```
Comment (dblock) [inline lib/iex/errors/invalid_option_error.rb]: * We should capture the actual option at least in the internals of the error (a field).
* `InvalidOptionError` is a generic name (as it should be) and you can't imply that it was called from get chart. So, the description should simply be `Invalid option` and additional detail would point on the error.

**Comment by dblock** [comment]
Looks like IEX returns a 400 on all kinds of errors, such as this one. I think you should create a `BadRequestError` and parse the returned body instead using the `error` field as the message or "Unexpected error" when that's not available + exposing parsed error details in the exception.

**Comment by rodolfobandeira** [comment]
At first, I thought that since this exception is only available through the `Chart.get`, it would be enough to assume that any 400 would be a result of invalid parameters. Thinking twice, what you said make sense. I'll review the logic and implement what you suggested. I'll update in here later this week. Thanks @dblock

**Comment by rodolfobandeira** [comment]
Hey @dblock, thanks for reviewing the code. I added two additional commits to address the points you suggested. Now we're filtering by that specific string `"option" is not allowed`. If it doesn't match that condition, we'll raise our own `BadRequestError`. The only problem now is figuring out how to write a propper RSpec to fake this edge case with "Bad Request". I found that adding no parameter gives an ArgumentError, and adding invalid data on valid parameters, returns an URI Exception. 

Please let me know your thoughts,
Cheers

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
-### 0.3.4 (Next)
+### 0.3.5 (Next)
 
 * Your contribution here.
 
+### 0.3.4 (2018/07/24)
```
Comment (dblock) [inline CHANGELOG.md]: We have not released 0.3.4, so undo that.

File: README.md [hunk 3]
```diff
@@ -146,6 +146,18 @@ See [#chart](https://iextrading.com/developer/docs/#chart) for detailed document
 
 If a symbol cannot be found an [IEX::Errors::SymbolNotFound](lib/iex/errors/symbol_not_found_error.rb) exception is raised.
 
+### InvalidOption
+
+If an invalid option is passed to `Chart.get`, an [IEX::Errors::InvalidOption](lib/iex/errors/invalid_option_error.rb) exception is raised.
```
Comment (dblock) [inline README.md]: I think it's enough to say "An IEX:...InvalidOption is raised when invalid options are specified."

File: README.md [hunk 4]
```diff
@@ -146,6 +146,18 @@ See [#chart](https://iextrading.com/developer/docs/#chart) for detailed document
...
+
+### BadRequest
+
+nop
```
Comment (dblock) [inline README.md]: Maybe "All 400 bad request responses from IEX result in a BadRequest."?

**Comment by rodolfobandeira** [comment]
Hey @dblock,  I agree with you. As long as it overrides the ugly Faraday exception, I believe it is something achieved. So I rolled-back the `InvalidOptionError` and left just the `BadRequestError`.

I'm gonna try working on some missing methods for the API to familiarize better with the project. 
Cheers

**Comment by rodolfobandeira** [comment]
Let's move this PR/thread to #9

**Pull request closed without merging**